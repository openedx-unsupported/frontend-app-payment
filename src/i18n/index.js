import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
