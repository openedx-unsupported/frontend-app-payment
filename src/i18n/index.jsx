import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
