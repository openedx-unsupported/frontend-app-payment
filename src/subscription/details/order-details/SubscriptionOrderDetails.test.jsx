import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import { SubscriptionOrderDetails } from './SubscriptionOrderDetails';

describe('SubscriptionOrderDetails', () => {
  it('should render skeleton when programTitle is empty string', () => {
    const tree = renderer.create(
      <IntlProvider locale="en">
        <SubscriptionOrderDetails programTitle="" />
      </IntlProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render order details for Dummy Program', () => {
    const tree = renderer.create(
      <IntlProvider locale="en">
        <SubscriptionOrderDetails programTitle="Dummy Program" />
      </IntlProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
