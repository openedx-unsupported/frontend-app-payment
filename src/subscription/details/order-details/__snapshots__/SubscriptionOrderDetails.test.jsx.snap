// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionOrderDetails should render order details for Dummy Program 1`] = `
<div
  className="basket-section order-details"
>
  <h5
    aria-level="2"
  >
    Order details
  </h5>
  <p
    className="mb-2"
  >
    The above total includes any applicable taxes.
  </p>
  <p
    className="mb-2"
  >
    After you complete your order you will have verified access to each course in Dummy Program.
  </p>
  <p>
    Cancel anytime.
  </p>
</div>
`;

exports[`SubscriptionOrderDetails should render skeleton when programTitle is empty string 1`] = `
Array [
  <div
    className="skeleton py-2 mb-3 w-50"
  />,
  <div
    className="skeleton py-2 mb-2 mr-4"
  />,
  <div
    className="skeleton py-2 mb-5 w-75"
  />,
]
`;
